### PR TITLE
Allow nullable types with generic arguments in directives

### PR DESCRIPTION
**Pull request by ajaybhargavb**
#1607 

@NTaylorMullen

### CODE REVIEW

File: src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs [hunk 1]
```diff
@@ -993,6 +993,7 @@ private void TypeArgumentList()
             Assert(CSharpSymbolType.LessThan);
             Balance(BalancingModes.None);
             Optional(CSharpSymbolType.GreaterThan);
+            Optional(CSharpSymbolType.QuestionMark);
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs]: This doesn't seem like it should be here. It's not part of a type argument list.

Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs]: What about types like `int?` do they have the same bug?

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs]: They don't. Types like `int?` are handled by https://github.com/aspnet/Razor/blob/dev/src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs#L962

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs]: > It's not part of a type argument list.

Sure. I can move this to the line after this method call.

Comment (pranavkm) [inline src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs]: Are generic arrays `Foo<int>[]` similarly covered?

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs]: Nice catch. Need to support that too.

**Comment by ajaybhargavb** [comment]
🆙 📅

**Pull request merged**